batch/batch/worker/worker.py

```diff
@@ -2033,6 +2073,7 @@ async def execute(self, classpath: str, scratch_dir: str, command_string: List[s
             elif message == JVM.FINISH_CANCELLED:
                 assert wait_for_interrupt.done()
                 log.info(f'{self}: was cancelled')
+                raise JobDeletedError()
```

Comment:
if you remove the parentheses then Python won't allocate an object if the exception is not allocated unless it is "needed". https://docs.python.org/3/reference/simple_stmts.html#the-raise-statement

Let's avoid the parens here.